backend/src/neuroagent/tools/circuit_population_analysis_tool.py

```diff
@@ -0,0 +1,335 @@
+- model_processing (VARCHAR): Processing approach (e.g., 'fullaxon', 'axon_bbpv5')
+
+Biological Properties:
+- population (VARCHAR): Name of the population this neuron belongs to
+- layer (VARCHAR): Cortical layer (e.g., 'L1', 'L2/3', 'L4', 'L5', 'L6')
+- cell_type (VARCHAR): Cell type classification (e.g., 'PYR', 'INT', 'SST', 'PV')
+- mtype (VARCHAR): Morphological type
+- etype (VARCHAR): Electrical type
+- region (VARCHAR): Brain region
+- subregion (VARCHAR): Brain subregion
+
+Additional Common Columns:
+- ei (VARCHAR): Excitatory ('e') or Inhibitory ('i') classification
+- synapse_class (VARCHAR): Synaptic class
+- depth (FLOAT): Depth within cortical column
+- tuning_angle (FLOAT): Orientation tuning preference
+- dynamics_params (VARCHAR): JSON string of parameter overrides
+
+Query Patterns for Neural Analysis:
+
+Cell Type Analysis:
+- Count neurons by model_type, cell_type, layer
+- Filter excitatory vs inhibitory neurons
+- Analyze morphological diversity
+
+Spatial Analysis:
+- Distance calculations using SQRT((x1-x2)^2 + (y1-y2)^2 + (z1-z2)^2)
+- Neurons within spatial boundaries
+- Layer-specific distributions
+
+Circuit Composition:
+- Population statistics by region/layer
+- Cell type distributions
+- Morphology usage patterns
+
+Model Properties:
+- Group by model_template or model_processing
+- Filter biophysical vs point neurons
+- Virtual neuron identification
+
+Convert neuroscience questions about circuit populations to SQL queries that analyze neuron properties, spatial distributions, cell types, morphologies, and circuit composition using the SONATA data format understanding."""
+
+                user_prompt = f"""Convert this neuroscience question about circuit population to a SQL SELECT query.
+
+Table: 'neurons' (circuit population data)
+Columns: {columns}
+Types: {dtypes}
+Sample neuron records: {sample}
+
+Question about neurons: {self.input_schema.question}
+
+Generate the SQL query to analyze the neuron population:"""
+
+                # Get SQL from OpenAI
+                model = "gpt-4o-mini"
+
+                response = await self.metadata.openai_client.chat.completions.create(
+                    model=model,
+                    messages=[
+                        {"role": "system", "content": system_prompt},
+                        {"role": "user", "content": user_prompt},
+                    ],
+                )
+
+                if response.choices[0].message.content:
+                    sql = response.choices[0].message.content.strip()
+                else:
+                    raise ValueError("No content in chat completion.")
+
+                # Clean up response (remove markdown if present)
+                if "```" in sql:
+                    parts = sql.split("```")
+                    for part in parts:
+                        if "SELECT" in part.upper():
+                            sql = part.replace("sql", "").strip()
+                            break
```

Comment:
Why don't we use structured output instead? The current

```
```programming language
```
```

parsing is exactly how one would do things before structured output existed.